backend/app/Http/Controllers/Api/SessionController.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Http\Controllers\Api;
+
+use App\Actions\Sessions\GetAllSessionsAction;
+use App\Http\Controllers\Controller;
+use App\Http\Response\Sessions\GetAllSessionsResponse;
+use App\Http\Response\ApiResponse;
+
+final class SessionController extends Controller
+{
+    private $getAllSessionsAction;
+
+    public function __construct(GetAllSessionsAction $getAllSessionsAction)
+    {
+        $this->getAllSessionsAction = $getAllSessionsAction;
+    }
+
+    public function getAllSessions()
+    {
+        $response = $this->getAllSessionsAction->execute();
+
+        return ApiResponse::success(
+            new GetAllSessionsResponse([
+                'sessions' => $response->sessions()
```

Comment:
you still passing array instead of object, pass `response` object or session collection, and format it inside `toArray`

Reply:
make it just `new GetAllSessionsResponse($response->sessions())` or `new GetAllSessionsResponse($response)` and convert to needed format in `toArray`, try to avoid using arrays for passing anything in constructors, that going to be used inside of it, since its responsibility of `GetAllSessionsResponse` to format response, and not controllers